lee_harvey/class_scripts/stocks_challenge.py

```diff
@@ -2,6 +2,46 @@
 
 print()
 
+def get_stock():
+    return input("Which stock are you interested in? \
+    \nType 'amzn' for Amazon, 'appl' for Apple, 'fb' for Facebook, 'goog' for Google and 'msft' for Microsoft. \n")
+
+def get_name():
+    return input("Please enter your name: ")
+
+def get_savings():
+    return int(input("Please enter the amount of savings you would like to work with: "))
+
+def evaluate(s):
+    # match s:  Requires Python 3.10 or higher
+    #     case 'amzn':
+    #         return amazon, 'Amazon'
+    #     case 'appl':
+    #         return apple, 'Apple'
+    #     case 'fb':
+    #         return fb, 'Facebook'
+    #     case 'goog':
+    #         return google, 'Google'
+    #     case 'msft':
+    #         return msft, 'Microsoft'
+    if s == 'amzn':
+        return amazon, 'Amazon'
+    elif s == 'appl':
+        return apple, 'Apple'
+    elif s == 'fb':
+        return fb, 'Facebook'
+    elif s == 'goog':
+        return google, 'Google'
+    elif s == 'msft':
+        return msft, 'Microsoft'
+    else:
+        s = get_stock()
```

Comment:
This will not work if the user enters the the incorrect stock name twice. You will get a `TypeError: cannot unpack non-iterable NoneType object` because `stock, s_name = evaluate(u_stock)` is looking for two values; however, if the user inputs incorrectly twice, then you will only get one variable 's' from `s = get_stock()`.

- You are on the right track with constructing a `while` statement to continue prompting the user to enter the right stock name.

- while loops execute a set of statements as long as a condition is true.

Reply:
Also I have already marked that you have completed this challenge so you don't have to continue working on it if you don't want to.